config/src/main/java/io/atomix/core/config/jackson/JacksonConfigProvider.java

```diff
@@ -179,4 +188,54 @@ public PrimitiveConfigDeserializer() {
       super(PrimitiveConfig.class, type -> PrimitiveTypes.getPrimitiveType(type).configClass());
     }
   }
+
+  private static class InterpolatingYamlFactory extends YAMLFactory {
+    @Override
+    protected YAMLParser _createParser(InputStream in, IOContext ctxt) throws IOException {
+      return new InterpolatingYamlParser(ctxt, _getBufferRecycler(), _parserFeatures, _yamlParserFeatures, _objectCodec, _createReader(in, null, ctxt));
+    }
+  }
+
+  private static class InterpolatingYamlParser extends YAMLParser {
+    private static final Pattern SYS_PATTERN = Pattern.compile("\\$\\{sys:([A-Za-z0-9-_.]+)\\}");
```

Comment:
I was thinking no static because it's bootstrap only then it may be garbage collected but maybe the jvm is smart enough.

Reply:
🤔

Reply:
nope.

Static variables cannot be elected for garbage collection while the class is loaded.

so I guess for it to happen you'd need to enable class unloading.